implementation/create_name_procedure.sql

```diff
@@ -0,0 +1,205 @@
+    @legal_alias_given_name_ipa AS NVARCHAR,            -- ipa for legal alias given name entered into the UI..
+
+    @legal_alias_family_name_unicode as NVARCHAR,       -- unicode for legal alias family name entered into the UI.
+    @legal_alias_family_name_latin1 as VARCHAR,         -- latin1 for legal alias family name entered into the UI.
+    @legal_alias_family_name_ipa AS NVARCHAR,           -- IPA for legal alias family name entered into the UI.
+
+    @preferred_locale_id as INT,                        -- passed by UI when locale preference selected on name create name form
+
+    -- ****** GENDER IDENTITY INFORMATION (see UI mockup) ****** --
+    @gender_identity AS VARCHAR,                        -- Gender identity entered via text field in the UI
+    @nominative_pronouns AS VARCHAR,                    -- Gender pronouns entered via text field in the UI
+    @genative_pronouns AS VARCHAR,                      -- Name Prefix - logically grouped with gender information
+    @honorific_text As VARCHAR, 
+
+    -- ****** CULTERAL IDENTITY INFORMATION (see UI mockup) ****** --
+    @tribe_clan_name_unicode  AS NVARCHAR,               -- text of tribe or clan unicode name enered into the UI.
+    @tribe_clan_name_latin1 as VARCHAR,                 -- text of tribe or clan unicode name enered into the UI.
+    @tribe_clan_name_ipa AS NVARCHAR,                   -- text of tribe or clan unicode name enered into the UI.
+
+    @mother_person_id as INT,                           -- passed if association is made on the name record, null if not
+    @father_person_id as INT,                           -- passed if association is made on the name record, null if not
+
+    -- ****** OVERRIDE IDENTITY INFORMATION (See UI mockup) ****** --
+    @override_full_name_unicode as NVARCHAR,           -- text of unicode full override name entered into the UI.
+    @override_full_name_latin1 as VARCHAR,             -- text of latin1 full override name entered into the UI.
+    @override_full_name_ipa AS NVARCHAR,               -- text for ipa of full override name value entered into the UI.
+
+)
+AS BEGIN
+    BEGIN TRANSACTION
+        BEGIN TRY
+            -- Orchestrate proper order of operations for inserting a name
+
+            -- Step 1: Call f_validate_email and throw error if email already in the database
+                -- if there is a match, throw an error
+                -- Piggy: No, if the email is already in the database, get the person_id that goes with it and use that going forward.
+                --        It is perfectly reasonable to create more than one name for a given person.
+
+                -- if no match, perform an insert into the persons table and capture new ID in a variable
+                -- variables created: @new_person_id
+            
+            -- Piggy: Rather than these validation functions, provide conditional insert functions that return the id.

```

Comment:
I think the upsert code in hw08 is a good model to follow. However, we want the ids back, so I think we want functions.